pkg/controllers/user/rbac/podsecuritypolicy/template.go

```diff
@@ -37,37 +38,41 @@ func RegisterTemplate(ctx context.Context, context *config.UserContext) {
 	}
 	clusterRoleInformer.AddIndexers(clusterRoleIndexer)
 
-	lfc := &Lifecycle{
-		policies:          context.Policy.PodSecurityPolicies(""),
-		policyLister:      context.Policy.PodSecurityPolicies("").Controller().Lister(),
-		clusterRoles:      context.RBAC.ClusterRoles(""),
-		clusterRoleLister: context.RBAC.ClusterRoles("").Controller().Lister(),
+	handler := &psptHandler{
+		policies:                   context.Policy.PodSecurityPolicies(""),
+		podSecurityPolicyTemplates: context.Management.Management.PodSecurityPolicyTemplates(""),
+		clusterRoles:               context.RBAC.ClusterRoles(""),
 
 		policyIndexer:      policyInformer.GetIndexer(),
 		clusterRoleIndexer: clusterRoleInformer.GetIndexer(),
 	}
 
-	pspti := context.Management.Management.PodSecurityPolicyTemplates("")
-	psptSync := v3.NewPodSecurityPolicyTemplateLifecycleAdapter("cluster-pspt-sync_"+context.ClusterName, true, pspti, lfc)
-	context.Management.Management.PodSecurityPolicyTemplates("").AddHandler(ctx, "pspt-sync", psptSync)
+	context.Management.Management.PodSecurityPolicyTemplates("").AddHandler(ctx, "pspt-sync", handler.sync)
 }
 
-type Lifecycle struct {
-	policies          v1beta12.PodSecurityPolicyInterface
-	policyLister      v1beta12.PodSecurityPolicyLister
-	clusterRoles      v12.ClusterRoleInterface
-	clusterRoleLister v12.ClusterRoleLister
+type psptHandler struct {
+	policies                   v1beta12.PodSecurityPolicyInterface
+	podSecurityPolicyTemplates v3.PodSecurityPolicyTemplateInterface
+	clusterRoles               v12.ClusterRoleInterface
 
 	policyIndexer      cache.Indexer
 	clusterRoleIndexer cache.Indexer
 }
 
-func (l *Lifecycle) Create(obj *v3.PodSecurityPolicyTemplate) (runtime.Object, error) {
-	return nil, nil
-}
+func (p *psptHandler) sync(key string, obj *v3.PodSecurityPolicyTemplate) (runtime.Object, error) {
+	if obj == nil {
+		return nil, nil
+	}
+
+	if _, ok := obj.Annotations[podSecurityPolicyTemplateCleanup]; !ok {
```

Comment:
nit: if it doesn't have cleanup annotation then cleanup? Cleaned would make more sense to me

Reply:
agree